docs/src/pages/presets/writing-presets/index.md

```diff
@@ -0,0 +1,120 @@
+---
+id: 'writing-presets'
+title: 'Writing Presets'
+---
+
+[Storybook presets](../introduction/) are grouped collections of `babel`, `webpack`, and `addons` configurations that support specific use cases in Storybook, such as typescript or MDX support.
+
+This doc covers the [presets API](#presets-api) and how to use the presets mechanism for [advanced configuration](#advanced-configuration).
+
+## Presets API
+
+A preset is a set of hooks that is called by Storybook on initialization and can override configurations for `babel`, `webpack`, `addons`, and `entries`.
+
+Each configuration has a similar signature, accepting a base configuration object and options, as in this webpack example:
+
+```js
+export async function webpack(baseConfig, options) {
+  // modify or replace config
```

Comment:
IMO the best practice is to make it immutable. Mutating the original reference object can cause unexpected bugs